docs/products/access/topics/oidc/claims-lifecycle/claims-lifecycle.md

```diff
@@ -0,0 +1,76 @@
+           AccessToken / IDToken pair. The IDToken is the primary source of information about the user. An internal User Object is created
+           and populated with all the user claims available in the IDToken. In addition to the user attributes, the AMR claim is also stored
+           in the User Object.
+        2. If the IdentityProvider is of type Tulip, it is likely that a User Data Enhancer (UDH) API integration is configured. If this is
+           the case, the user claims are requested via another API call from UDH, Access passes the user identifier (subject) and a set of
+           scopes requested by the client. Otherwise, the User Object is persisted (see #2.a.5).
+        3. The UDH response is mapped to a UserInfo standard format.
+        4. The User Object is populated with any claims returned by the UDH service. If a given claim already exists in the user object,
+           it's overwritten with the one coming from UDH, which is considered more recent.
+        5. The User Object with all the claims is persisted.
+    2. If the underlying IdentityProvider is of SAML type (e.g. CIM), Access redirects the user to the defined SAML Identity Provider with
+       SAML AuthnRequest (whether it is a redirect or a POST call depends on the integration characteristic).
+        1. Access processes the SAML AuthnResponse.
+           SAML [Attributes Mapping](https://docs.onegini.com/products/access/topics/general-app-config/identity-providers/identity-providers/#attribute-mapping)
+           is triggered if any are defined for a given SAML Identity Provider. If no mappings are available all SAML assertion attributes
+           are used in the next step.
+        2. SAML assertion attributes are mapped to claims according to the configuration. Any unmapped attributes are discarded/filtered
+           out.
+        3. An internal user object is created and populated with all user claims provided by the SAML assertion. Next to the user
+           attributes, the ARR (AMR) assertion is also stored in the User Object.
+        4. If the IdentityProvider is of type CIM, it is likely that a Person API integration is configured. If this is the case, the Person
+           profile is requested by another API call from the CIM, Access will pass the person identifier. Otherwise, the User object will be
+           persisted (see [#2.2.7.](#2.2.7.)).
+        5. The Person API response is mapped to a standard UserInfo format.
+        6. The User object is populated with any claims returned by the CIM Person API. If a given claim is already present in the User
+           object, it's overwritten with the one coming from the CIM Person Profile, which is considered more recent.
+        7. The User Object with all the claims is persisted.
+3. The authorization request is considered successful, Access generates the _authorization_code_.
```

Comment:
To make it even more complex, Access Token Customisation Web Hook is called in Authorization request to fetch scopes
`AuthorizationCodeGrantRequestHandler#getRequestedScopes`
and user claims are also resolved (because we have user claims in CustomizeAccessTokenRequest).

Reply:
But maybe we can omit this as an implementation detail not to obscure the charts.